### PR TITLE
Updates to v0.2.1

### PR DESCRIPTION
**Pull request by msahraeian**
Multiple pull request and updates:

- 2x-3x speed improvement for `scan_alignment`
- Removed MD tag requirement for bam files
- More program options added for `scan_alignment`
- Option for normalizing MQ, BQ, and other alignment channels for input candidate matrices
- New pre-trained models added based on SEQC-II dataset
- Minor fixes
- Update to Pytorch 1.1.0,  torchvision 0.3.0, and cudatoolkit 9.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -236,20 +234,25 @@ You can then used the synthetic tumor/normal pair and the known *in silico* spik
 
 ## Trained Network Models
 We provide a set of trained NeuSomatic network models for general purpose usage. Users should note that these models are trained for sepcific settings and are not supposed to work perfectly for all circumestances.
+
+The SEQC-II pretrained models are the recommended NeuSomatic models and are analyzed in details in [Sahraeian et al. 2019](https://doi.org/10.1101/667261).
```
Comment (marghoob) [inline README.md]: details --> detail

File: neusomatic/include/msa.hpp [hunk 2]
```diff
@@ -64,29 +44,40 @@ class GappedSeq {
...
         }
       }
       result += bases_[i];
     }
     return result;
   }
+
+  int length() const {
+    int res = 0;
+    for (size_t i = 0; i < bases_.size(); ++i) {
```
Comment (marghoob) [inline neusomatic/include/msa.hpp]: Do `bases_` and `gaps_` have the same size here? Can you not have `i` loop condition depend on `gaps_.size()`?

**Comment by marghoob** [comment]
👍 If @roger-liu-bina hasn't reviewed this, please check with him too.

File: neusomatic/include/msa.hpp (hunk 2 above)
Comment (msahraeian) [reply neusomatic/include/msa.hpp]: that's @roger-liu-bina 's code.
@roger-liu-bina would you please comment?

**Comment by msahraeian** [comment]
The scan_alignment c++ codes are from @roger-liu-bina pull requests #35 and #43. I have included them in this pull request. 

@roger-liu-bina would you please review this pull request.

File: neusomatic/include/msa.hpp (hunk 2 above)
Comment (roger-liu-bina) [reply neusomatic/include/msa.hpp]: @marghoob Yes `bases_` and `gaps_` always have the same size.

**Comment by roger-liu-bina** [comment]
Looks good. The cpp things have been tested by me and @msahraeian carefully. The python things are most reformating I think. There are some duplicated vars but not a big deal. https://github.com/bioinform/neusomatic/pull/44/files#diff-7088a29d6c9d09e4d21a0860a969fb27L57

**Comment by marghoob** [comment]
Thanks @roger-liu-bina . I agree, we are good to go with this one.
